src/arrayMethodJoin.js

```diff
@@ -5,7 +5,34 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let result = '';
+    let currentSeparator = separator;
```

Comment:
You aren't gonna mutate `separator`, so you don't need to create a separate variable for it